lib/Dialect/FIRRTL/LowerTypes.cpp

```diff
@@ -0,0 +1,459 @@
+//===- LowerTypes.cpp - Lower FIRRTL types to ground types ----------------===//
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/FIRRTL/Ops.h"
+#include "circt/Dialect/FIRRTL/Passes.h"
+#include "circt/Dialect/FIRRTL/Types.h"
+#include "circt/Dialect/FIRRTL/Visitors.h"
+#include "circt/Support/LLVM.h"
+#include "llvm/ADT/StringMap.h"
+using namespace circt;
+using namespace firrtl;
+
+static Type getPortType(FlatBundleFieldEntry field) {
+  return field.isOutput && field.type.isa<FIRRTLType>()
+             ? FlipType::get(field.type.dyn_cast<FIRRTLType>())
```

Comment:
This should be `field.type.cast` instead of `dyn_cast` since you know it always succeeds